### PR TITLE
feat(SF-1152): Refactor pastPurchase saga to behave more like search

### PR DESCRIPTION
**Pull request by Vic-Dev**

### CODE REVIEW

File: src/core/sagas/recommendations.ts [hunk 1]
```diff
@@ -109,48 +111,61 @@ export namespace Tasks {
...
+        });
+        const results = yield effects.call(Requests.search, flux, request);
+        const navigations = PastPurchaseAdapter.pastPurchaseNavigations(
+          yield effects.select(Selectors.config),
+          SearchAdapter.combineNavigations(results)
+        );
+        yield effects.put(<any>flux.actions.receivePastPurchaseRefinements(navigations));
+      }
+    } catch (e) {
+      return effects.put(flux.actions.receivePastPurchaseRefinements(e));
```
Comment (Vic-Dev) [inline src/core/sagas/recommendations.ts]: Should I add this action to the undo error actions? @jmbtutor

Comment (jmbtutor) [reply src/core/sagas/recommendations.ts]: If it affects the state, then yes, which it looks like it does.

File: test/unit/core/sagas/recommendations.ts [hunk 2]
```diff
@@ -375,50 +375,78 @@ suite('recommendations saga', ({ expect, spy, stub }) => {
...
-        // });
+      it('should return if there are no pastPurchaseSkus', () => {
+        const flux: any = {};
+        const action: any = {};
+        const config = {};
+
+        const task = Tasks.fetchPastPurchaseNavigations(flux, action);
+
+        expect(task.next().value).to.eql(effects.select(Selectors.pastPurchases));
+        expect(task.next([]).value).to.be.undefined;
```
Comment (jmbtutor) [inline test/unit/core/sagas/recommendations.ts]: It's better to test that `.done` is `true` if you want to check that a generator has returned.

File: src/core/sagas/recommendations.ts (hunk 1 above)
Comment (Vic-Dev) [reply src/core/sagas/recommendations.ts]: just kidding it's already there

**Pull request merged**